### PR TITLE
resolve @types/react from the react env

### PR DESCRIPTION
**Pull request by KutnerUri**
## Proposed Changes

- always resolve `@types/react` relative to the env

This PR is pretty safe, because every react environment that uses typescript also adds @types/react as deps on the package, but we still should test it in a variety of edge cases.

TODO:
- [x] Wait for #5443 (depends on it to work correctly)
- [x] test on base-ui scope (without env)
- [x] test on design scope (with custom env)
- [ ] test on bit-dev (with many envs)
- [ ] verify tsconfig logic makes sense

### CODE REVIEW

**Comment by GiladShoham** [comment]
@KutnerUri don't you have to use baseUrl as well?
according to [ts docs](https://www.typescriptlang.org/tsconfig#paths)

**Comment by KutnerUri** [comment]
Apparently not - it's just when the path is not relative, ie 'node_modules/@types/react'.
It seems to work without it when using './node_modules/@types/react'

File: scopes/react/react/typescript/tsconfig.json [hunk 1]
```diff
@@ -11,7 +11,12 @@
     "skipLibCheck": true,
     "moduleResolution": "node",
     "esModuleInterop": true,
-    "resolveJsonModule": true
+    "resolveJsonModule": true,
+    "paths": {
+      "react": [
+        "./node_modules/@types/react"
+      ]
```
Comment (KutnerUri) [inline scopes/react/react/typescript/tsconfig.json:18]: not sure about this rule - need to check

File: scopes/react/react/typescript/tsconfig.build.json [hunk 2]
```diff
@@ -15,7 +15,13 @@
     "moduleResolution": "node",
     "esModuleInterop": true,
     "rootDir": ".",
-    "resolveJsonModule": true
+    "resolveJsonModule": true,
+    "paths": {
+      "react": [
+        "./node_modules/@types/react",
+        "../node_modules/@types/react"
+      ]
```
Comment (KutnerUri) [inline scopes/react/react/typescript/tsconfig.build.json:23]: this seems stable - it tells TS to look for `react` at the current folder `node_modules` or the parent `node_modules`.

this makes sense in the current capsule structure:
```
shared dir
- node_modules <-- shared modules
-- @types 
--- react <-- resolved location (fallback)
- target-dir
-- tsconfig.json (the '.' path)
-- node_modules
--- @types
---- react <-- resolved location (preferred)
```

Comment (KutnerUri) [reply scopes/react/react/typescript/tsconfig.build.json:23]: real world example:
Screen Shot 2022-05-08 at 20 36 29

**Comment by KutnerUri** [comment]
@GiladShoham - FYI we can merge it now. Seems less critical because Facebook fixes the `@types/react` somehow

**Review by ranm8 — COMMENTED** [review]
test
